### PR TITLE
New n3-utils and error-utils updater libraries

### PR DESCRIPTION
**Pull request by glenna-m**

### CODE REVIEW

File: components/lib/n3-utils.js [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+/**
+ * Converts an N3 store into TTL, loads the TTL, and converts the TTL to JSON-LD.  This function
+ * is intended primarily for testing/double checking the n3ToJsonld method and will likely be 
+ * deleted or deprecated in the future. 
+ *
+ * @param store    (required) the N3 store to convert to text/Turtle format
+ * @param frame    (optional) a JSON-LD frame
+ * @param filename (optional) an optional file name to write the Turtle result into.
+ *
+ * @return a promise to return the N3 store converted to JSON-LD format or a reject on error
```
Comment (dbooth-boston) [inline components/lib/n3-utils.js]: Suggest rewording this: @return a promise to return either the JSON-LD format or a reject on error

File: components/lib/n3-utils.js [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+}
+
+/**
+ * Converts an N3 store into JSON-LD. It is an asychronous function.
+ * 
+ * @param store    (required) the N3 store to convert to JSON-LD format
+ * @param frame    (required) a JSON-LD frame
+ * @param filename (optional) an optional file name to write the Turtle result into.
+ *
+ * @return a promise to return the N3 store converted to JSON-LD format or a reject on error
```
Comment (dbooth-boston) [inline components/lib/n3-utils.js]: Suggest rewording this to: @return a promise to return either the resulting JSON-LD or a reject on error

File: components/lib/n3-utils.js [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+ */ 
+function n3ToJsonld(n3Store, frame, filename) {
+
```
Comment (dbooth-boston) [inline components/lib/n3-utils.js]: Small stylistic request: I prefer having no blank line after the function header.  (To my eyes it helps keep the function together and it helps make the code fit in the visible window on the screen.)  If you think it is better to have one then let's take a poll of our developers to determine a preference.  It is not covered in the google style guide.

**Review by dbooth-boston — CHANGES_REQUESTED** [review]
LGTM subject to very minor comments.  No need for re-review.

**Review by dbooth-boston — APPROVED** [review]
LGTM

File: components/lib/n3-utils.js [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+                    function(dataset, triple) {
+                        var graph = triple.graph || '@default';
+                        dataset[graph] = dataset[graph] || [];
+                        dataset[graph].push({
+                            subject: fromN3(triple.subject),
+                            predicate: fromN3(triple.predicate),
+                            object: fromN3(triple.object)
+                        });
+                        return dataset;
+                    }, {'@default': []}), {})
```
Comment (dmtaub) [inline components/lib/n3-utils.js:222]: took me a moment to realize that {} was a second argument to a function above...

File: test/data/cmumps-patient7.jsonld [hunk 5]
```diff
@@ -3380,7 +3380,7 @@
             "label": "Periodic comprehensive preventive medicine reevaluation and management of an individual including an age and gender appropriate history, examination, counseling/anticipatory guidance/risk factor reduction interventions, and the ordering of appropriat",
             "patient": {
                 "id": "Patient-000007",
-                "label": "BUNNY, BUGS"
```
Comment (dmtaub) [inline test/data/cmumps-patient7.jsonld:3383]: Remind me: is this because the test output didn't match the test input?

File: test/error-utils-mocha.js [hunk 6]
```diff
@@ -0,0 +1,123 @@
+// error-utils-mocha.js
```
Comment (dmtaub) [inline test/error-utils-mocha.js:1]: Yay tests!

**Comment by dmtaub** [comment]
LGTM
